src/makeInfinityAdder.js

```diff
@@ -1,12 +1,35 @@
 'use strict';
 
 /***
- *
  * @return {function}
  */
-
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(value) {
+    if (arguments.length === 0) {
+      return sum;
+    }
+
+    sum += value;
+
+    function innerAdder(nextValue) {
+      if (arguments.length === 0) {
+        const currentSum = sum;
+
+        sum = 0;
+
+        return currentSum;
+      }
+      sum += nextValue;
+
+      return innerAdder;
+    }
```

Comment:
The innerAdder function should also return itself to allow for continued chaining after each call with a number.